### DISCUSSION TITLE
Tacspeak Ready or Not Voice Commands Mappings

### DISCUSSION THREAD

**Discussion by Devoutly1685** [jwebmeister/tacspeak#16] [Show and tell]
I'm using Tacspeak for the first time and decided to make this spreadsheet that shows the default voice command mapping. I'm hoping this will help new players to learn what voice commands are available by default.

The mapping can be found in the folder you have Tacspeak installed:
Tacspeak-3159-0-1-4-1703143447\tacspeak\grammar\ _readyornot.py

This is based on Tacspeak v0.1.4 and will be updated as new versions of Tacspeak are released.
https://docs.google.com/spreadsheets/d/1QbTS5vm1enHUQYcslKoCf2hVfdmsKXKzRxjWeG9nSlI/edit?usp=sharing

**VOICE COMMAND**: Phrases used to execute commands
**GAME COMMAND**: Command that will be executed
**TACSPEAK MAPPING**: Name of the mapping in the _readyornot.py config file
**CATEGORY**: Category of the mapping within the config file	
**DESCRIPTION**: Describes the in game actions of the command
**RON SEQUENCE**: Voice commands are converted to these sequences

**Comment by jwebmeister** [comment]
This is a good idea, I've just been too busy to collate everything into a user friendly format.

I've created a google sheet [here](https://docs.google.com/spreadsheets/d/1jpuR8JHmh0LOOcUQ7JMMzDOmSYYe2uMpy63X238ZySs/edit?usp=drive_link) with the most of the current commands + additional commands not in yet or that have since been removed. 
I made it for the primary purpose of recording audio / training data for model development, so there may be some misalignment between the commands in the google sheet and what's actually available in `./tacspeak/grammar/_readyornot.py`, so consulting the actual code is still best.  
To be clear, I'm not committing to maintaining or updating this google sheet, not until I'm certain the commands in Ready or Not are stable and unlikely to change (in regards to both Tacspeak and in-game).

Additionally there are options within the later versions of Tacspeak to dump all the active commands into a .txt file that might be useful for figuring out what commands are available, though it still is not better than reviewing the actual code, in my opinion. 
I've uploaded an example of the dumped .txt onto my google drive [here](https://drive.google.com/file/d/1OKBZkZ-GgxVRF8uxa5D9fvUabSZVhToK/view?usp=drive_link).

To dump the active commands, in `./tacspeak/user_settings.py` set:
- `DEBUG_MODE` = `True`
- `DEBUG_HEAVY_DUMP_GRAMMAR` = `True`

, then start `tacspeak.exe`. It'll generate a .txt file with the commands in it.  

Be sure to revert the settings back: 
- `DEBUG_MODE` = `False`
- `DEBUG_HEAVY_DUMP_GRAMMAR` = `False`

, and restart `tacspeak.exe`, otherwise you will find no commands register in-game (with debug mode on).
